.github/workflows/build.yml

```diff
@@ -16,83 +16,91 @@ jobs:
     name: Build jupyter_chat
 
     steps:
-    - name: Checkout
-      uses: actions/checkout@v3
+      - name: Checkout
+        uses: actions/checkout@v3
 
-    - name: Base Setup
-      uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
 
-    - name: Install dependencies
-      run: python -m pip install -U "jupyterlab>=4.0.0,<5"
+      - name: Install dependencies
+        run: python -m pip install -U "jupyterlab>=4.0.0,<5"
 
-    - name: Lint the packages
-      run: |
-        set -eux
-        jlpm
-        jlpm run lint:check
+      - name: Lint the packages
+        run: |
+          set -eux
+          jlpm
+          jlpm run lint:check
 
-    - name: Build the core package
-      run: jlpm build:core
+      - name: Build the core package
+        run: jlpm build:core
 
-    - name: Test the packages
-      run: |
-        set -eux
-        jlpm run test
+      - name: Test the packages
+        run: |
+          set -eux
+          jlpm run test
 
   test_extensions:
+    name: Python unit tests (Python ${{ matrix.python-version }})
     runs-on: ubuntu-latest
     needs: build_jupyter-chat
-    name: Python test on extensions
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - python-version: '3.9'
+          - python-version: '3.12'
```

Comment:
These are the only effective changes to the workflows. All other changes just fix the indentation of this YML file, and were performed automatically by my IDE's linter.